rotkehlchen/chain/evm/node_inquirer.py

```diff
@@ -1169,20 +1183,62 @@ def get_erc20_contract_info(self, address: ChecksumEvmAddress) -> dict[str, Any]
             contract = EvmContract(address=address, abi=abi, deployed_block=0)
             decoded = self._process_contract_info(
                 output=output,
-                properties=properties,
+                properties=ERC20_PROPERTIES,
                 contract=contract,
                 token_kind=EvmTokenKind.ERC20,
             )
             log.debug(f'{address} was succesfuly decoded as ERC20 token')
 
-        for prop, value in zip_longest(properties, decoded):
+        for prop, value in zip_longest(ERC20_PROPERTIES, decoded):
             if isinstance(value, bytes):
                 value = value.rstrip(b'\x00').decode()  # noqa: PLW2901
             info[prop] = value
 
-        self.contract_info_erc20_cache[address] = info
         return info
 
+    def _query_token_contract(
+            self,
+            token_abi_str: Literal['ERC20_TOKEN', 'ERC721_TOKEN'],
+            properties: tuple[str, ...],
+            address: ChecksumEvmAddress,
+    ) -> list[tuple[bool, bytes]]:
+        abi = self.contracts.abi(token_abi_str)
+        contract = EvmContract(address=address, abi=abi, deployed_block=0)
+        try:
+            # Output contains call status and result
+            output = self.multicall_2(
+                require_success=False,
+                calls=[(address, contract.encode(method_name=prop)) for prop in properties],
+            )
+        except RemoteError:
+            # If something happens in the connection the output should have
+            # the same length as the tuple of properties
+            output = [(False, b'')] * len(properties)
+
+        return output
+
+    def get_erc20_contract_info(self, address: ChecksumEvmAddress) -> dict[str, Any]:
+        """
+        Query an erc20 contract address and return basic information as:
+        - Decimals
+        - name
+        - symbol
+        At all times, the dictionary returned contains the keys; decimals, name & symbol.
+        Although the values might be None.
+        if it is provided in the contract. This method may raise:
+        - BadFunctionCallOutput: If there is an error calling a bad address
+        """
+        cache = self.contract_info_erc20_cache.get(address)
+        if cache is not None:
+            return cache
```

Comment:
nitpick: in those cases the walrus opeartor is nice

```suggestion
        if (cache :=  self.contract_info_erc20_cache.get(address)) is not None:
            return cache
```

Reply:
done